### PR TITLE
Add CCS hint header

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

Add an HTTP header to the request containing the user's id or login hint.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/IdentityCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1025,8 +1025,8 @@
 		B26CEB022367B3B9009E6E54 /* MSIDSystemWebViewControllerFactory.h in Headers */ = {isa = PBXBuildFile; fileRef = B26CEB002367B3B9009E6E54 /* MSIDSystemWebViewControllerFactory.h */; };
```
Comment (kaisong1990) [inline IdentityCore/IdentityCore.xcodeproj/project.pbxproj:1025]: This pull request does not update changelog.txt.

Please consider if this change would be noticeable to a partner or user and either update changelog.txt or resolve this conversation.

File: IdentityCore/src/MSIDConstants.m [hunk 2]
```diff
@@ -31,6 +31,7 @@
 NSString *const MSID_DEVICE_MODEL_KEY              = @"x-client-DM";
 NSString *const MSID_APP_NAME_KEY                  = @"x-app-name";
 NSString *const MSID_APP_VER_KEY                   = @"x-app-ver";
+NSString *const MSID_CCS_HINT_KEY                  = @"X-AnchorMailbox";
```
Comment (oldalton) [inline IdentityCore/src/MSIDConstants.m:34]: Interesting name for an auth parameter.... is this right?

File: IdentityCore/src/parameters/MSIDRequestParameters.m [hunk 3]
```diff
@@ -239,6 +239,35 @@ - (MSIDConfiguration *)msidConfiguration
...
+    NSString *utid = accountIdentifier.utid;
+    
+    if (![NSString msidIsStringNilOrBlank:uid] && ![NSString msidIsStringNilOrBlank:utid])
+    {
+        NSString *oidHeader = [NSString stringWithFormat:@"Oid:%@@%@", uid, utid];
+        appRequestMetadata[MSID_CCS_HINT_KEY] = oidHeader;
+    }
+    else if (![NSString msidIsStringNilOrBlank:accountIdentifier.displayableId])
+    {
+        NSString *upnHeader = [NSString stringWithFormat:@"UPN:%@", accountIdentifier.displayableId];
```
Comment (oldalton) [inline IdentityCore/src/parameters/MSIDRequestParameters.m:264]: seems somewhat duplicate to what is done in the interactive parameters. Possible to combine?

File: IdentityCore/src/requests/MSIDSilentTokenRequest.m [hunk 4]
```diff
@@ -102,6 +102,8 @@ - (void)executeRequestWithCompletion:(MSIDRequestCompletionBlock)completionBlock
         return;
     }
     
+    [self.requestParameters updateAppRequestMetadata:nil];
```
Comment (oldalton) [inline IdentityCore/src/requests/MSIDSilentTokenRequest.m:105]: Will we need to do the same for broker requests? Possible to have it done somewhere globally?

File: IdentityCore/src/MSIDConstants.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/src/MSIDConstants.m:34]: yep, it is correct :)

File: IdentityCore/src/parameters/MSIDRequestParameters.m (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/src/parameters/MSIDRequestParameters.m]: refactored

File: IdentityCore/src/requests/MSIDSilentTokenRequest.m (hunk 4 above)
Comment (antrix1989) [reply IdentityCore/src/requests/MSIDSilentTokenRequest.m]: Refactored, no changes needed on the broker side now.

File: IdentityCore/src/parameters/MSIDRequestParameters.m [hunk 5]
```diff
@@ -239,6 +249,34 @@ - (MSIDConfiguration *)msidConfiguration
...
+{
+    MSIDAccountIdentifier *accountIdentifier = self.accountIdentifier;
+    
+    if (![NSString msidIsStringNilOrBlank:homeAccountId])
+    {
+        accountIdentifier = [[MSIDAccountIdentifier alloc] initWithDisplayableId:nil homeAccountId:homeAccountId];
+    }
+    
+    NSMutableDictionary *appRequestMetadata = [self.appRequestMetadata mutableCopy];
+    [appRequestMetadata removeObjectForKey:MSID_CCS_HINT_KEY];
```
Comment (oldalton) [inline IdentityCore/src/parameters/MSIDRequestParameters.m:262]: Should we ensure that either uid/utid or upn is present before cleaning up this key?

Comment (oldalton) [reply IdentityCore/src/parameters/MSIDRequestParameters.m:262]: sounds good to me.

**Pull request merged**